server/channels/app/status.go

```diff
@@ -77,6 +77,13 @@ func (a *App) SetCustomStatus(c request.CTX, userID string, cs *model.CustomStat
 		return model.NewAppError("SetCustomStatus", "api.custom_status.set_custom_statuses.update.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	// Ensure the emoji exists before saving the custom status even if it's deleted afterwards
+	if cs.Emoji != "" {
```

Comment:
Could we simplify here?
```suggestion
	if cs.Emoji != "" && a.confirmEmojiExists(c, cs.Emoji) != nil  {
```

Unless your intention was to wrap the error below.

Reply:
Good catch. I split it up because the previous function I was using had multiple return parameters. I'll fix that

Reply:
Oh, I did want to wrap the error. I'll change that instead 😛